WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -1454,3 +1469,37 @@ def sendPilotAccounting(self, pilotDict):
       return result
 
     return S_OK()
+
+  def sendPilotSubmissionAccounting(self,
+                                    siteName,
+                                    ceName,
+                                    queueName,
+                                    numTotal,
+                                    numSucceeded,
+                                    status):
+
+    """ Send pilot submission accounting record
+    """
+
+    pA = PilotSubmissionAccounting()
+    pA.setStartTime(dateTime())
+    pA.setEndTime(dateTime())
+    pA.setValueByKey('HostName', DIRAC.siteName())
+    pA.setValueByKey('SiteDirector', self.am_getModuleParam('agentName'))
+    pA.setValueByKey('Site', siteName)
+    pA.setValueByKey('CE', ceName)
+    pA.setValueByKey('Queue', queueName)
+    pA.setValueByKey('Status', status)
+    pA.setValueByKey('NumTotal', numTotal)
+    pA.setValueByKey('NumSucceeded', numSucceeded)
+    result = gDataStoreClient.addRegister(pA)
+
+    if not result['OK']:
+      self.log.warn('Error in add Register:' + result['Message'])
+      return result
+
+    self.log.verbose("Begin commit")
+    result = gDataStoreClient.commit()
```

Comment:
This is inefficient, because every time you call this method you send the records to the Accounting system. 
I would propose at least use the delayed commit of the DataStoreClient (I can not remember the method name. please check it).